app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
+          #add the owner back to the member_ids array
+          mi_array.unshift(@application.owner.owner_ids[0])
```

Comment:
What is going on in these two lines?

Reply:
The owner cannot be removed, so I am removing the owner before editing and putting the owner back in the array before updating. The owner is always the first item in the member_ids, so I'm putting it back where it was.